sys/trickle/trickle.c

```diff
@@ -33,25 +35,21 @@ void trickle_callback(trickle_t *trickle)
 
 void trickle_interval(trickle_t *trickle)
 {
-    uint32_t old_interval = trickle->I;
-    uint32_t max_interval = trickle->Imin << trickle->Imax;
-    uint32_t diff = old_interval - trickle->t;
+    assert(trickle->I > 0);
 
-    trickle->I *= 2;
+    uint32_t diff = trickle->I - trickle->t;
 
-    if ((trickle->I == 0) || (trickle->I > max_interval)) {
-        trickle->I = max_interval;
-    }
+    trickle->I *= 2;
+    trickle->I = MIN(trickle->I, (trickle->Imin << trickle->Imax));
```

Comment:
I don't think its a corner case, if wrap over is possible than either the datatype of `trickle->I` is too small or the config params are messed up. IMHO it **must** hold that `(Imin << Imax) < UINT32_MAX/2` otherwise trickle interval is not bounded, checking for `0` doesn't help - btw. for debugging I added an `assert`.